api/v3/AdhocCharges.php

```diff
@@ -38,6 +38,9 @@ function civicrm_api3_adhoc_charges_create($params) {
  * @throws API_Exception
  */
 function civicrm_api3_adhoc_charges_get($params) {
+  if(empty($params['options']['limit'])){
+    $params['options'] = array('limit' => 0);
```

Comment:
it should be :

`$params['options']['limit'] = 0;`

because like that you are clearing the whole options array content  and replace it with array('limit' => 0) only in case there is not limit specified.